Foundation/NSArray.swift

```diff
@@ -247,99 +229,111 @@ open class NSArray : NSObject, NSCopying, NSMutableCopying, NSSecureCoding, NSCo
         return result
     }
     
-    open func firstObjectCommon(with otherArray: [AnyObject]) -> AnyObject? {
-        let set = NSSet(array: otherArray)
+    open func firstObjectCommon(with otherArray: [Any]) -> Any? {
+        let set = otherArray.map { _SwiftValue.store($0) }
 
         for idx in 0..<count {
-            let item = self[idx]
+            let item = _SwiftValue.store(self[idx])
             if set.contains(item) {
-                return item
+                return _SwiftValue.fetch(item)
             }
         }
         return nil
     }
 
-    internal func getObjects(_ objects: inout [AnyObject], range: NSRange) {
+    internal func getObjects(_ objects: inout [Any], range: NSRange) {
         objects.reserveCapacity(objects.count + range.length)
 
         if type(of: self) === NSArray.self || type(of: self) === NSMutableArray.self {
-            objects += _storage[range.toRange()!]
+            objects += _storage[range.toRange()!].map { _SwiftValue.fetch($0) }
             return
         }
         
-        
-        
         objects += range.toCountableRange()!.map { self[$0] }
     }
     
-    open func index(of anObject: AnyObject) -> Int {
+    open func index(of anObject: Any) -> Int {
+        guard let val = anObject as? AnyHashable else {
+            return NSNotFound
+        }
         for idx in 0..<count {
-            let obj = object(at: idx) as! NSObject
-            if anObject === obj || obj.isEqual(anObject) {
-                return idx
+            if let obj = object(at: idx) as? AnyHashable {
+                if val == obj {
+                    return idx
+                }
             }
         }
         return NSNotFound
     }
     
-    open func index(of anObject: AnyObject, in range: NSRange) -> Int {
+    open func index(of anObject: Any, in range: NSRange) -> Int {
+        guard let val = anObject as? AnyHashable else {
```

Comment:
Same as the AnyHashable/Equatable comment above (marking all of these for easier reference later, if need be)